campaigns_api.go

```diff
@@ -0,0 +1,146 @@
+	path := fmt.Sprintf("campaigns/%s/listsandsegments.json", url.QueryEscape(campaignID))
+	var ls campaigns.ListsAndSegments
+	err := c.client.Get(path, &ls)
+	return ls, err
+}
+
+func (c *campaignAPI) Recipients(campaignID string, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) (campaigns.Recipients, error) {
+	var r campaigns.Recipients
+	if orderField == orderfield.Date {
+		return r, errors.New("date is not a valid order field")
+	}
+
+	path := resultsPath("bounces", campaignID, time.Time{}, page, pageSize, orderField, orderDirection)
+	err := c.client.Get(path, &r)
+	return r, err
+}
+
+func (c *campaignAPI) Bounces(campaignID string, date time.Time, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) (campaigns.Bounces, error) {
+	path := resultsPath("bounces", campaignID, date, page, pageSize, orderField, orderDirection)
+	var b campaigns.Bounces
+	err := c.client.Get(path, &b)
+	return b, err
+}
+
+func (c *campaignAPI) Opens(campaignID string, date time.Time, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) (campaigns.CampaignRecipientActions, error) {
+	path := resultsPath("opens", campaignID, date, page, pageSize, orderField, orderDirection)
+	var o campaigns.CampaignRecipientActions
+	err := c.client.Get(path, &o)
+	return o, err
+}
+
+func (c *campaignAPI) Clicks(campaignID string, date time.Time, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) (campaigns.CampaignRecipientActions, error) {
+	path := resultsPath("clicks", campaignID, date, page, pageSize, orderField, orderDirection)
+	var o campaigns.CampaignRecipientActions
+	err := c.client.Get(path, &o)
+	return o, err
+}
+
+func (c *campaignAPI) Unsubscribes(campaignID string, date time.Time, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) (campaigns.Unsubscribes, error) {
+	path := resultsPath("unsubscribes", campaignID, date, page, pageSize, orderField, orderDirection)
+	var u campaigns.Unsubscribes
+	err := c.client.Get(path, &u)
+	return u, err
+}
+
+func (c *campaignAPI) SpamComplaints(campaignID string, date time.Time, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) (campaigns.SpamComplaints, error) {
+	path := resultsPath("spam", campaignID, date, page, pageSize, orderField, orderDirection)
+	var s campaigns.SpamComplaints
+	err := c.client.Get(path, &s)
+	return s, err
+}
+
+func (c *campaignAPI) Delete(campaignID string) error {
+	return c.client.Delete(fmt.Sprintf("campaigns/%s.json", campaignID))
+}
+
+func (c *campaignAPI) Unschedule(campaignID string) error {
+	return c.client.Delete(fmt.Sprintf("campaigns/%s/unschedule.json", campaignID))
+}
+
+func resultsPath(action string, campaignID string, date time.Time, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) string {
```

Comment:
`resultsPath` is generic and unclear in my opinion. Maybe something like `getRecipientActivityPath`?